client/blocks/jetpack-benefits/test/scan-history.jsx

```diff
@@ -30,43 +26,59 @@ describe( 'Scan History Jetpack Benefit Block', () => {
 	test( 'When the scan state is actively scanning, show a scanning message', () => {
 		const scanState = { state: 'scanning' };
 		getSiteScanState.mockReturnValue( scanState );
-		expect( getBenefitsCard().props() ).toHaveProperty( 'jestMarker', 'scanning' );
+
+		render( <JetpackBenefitsScanHistory /> );
+		const card = screen.getByTestId( 'scanning' );
+
+		expect( card ).toBeInTheDocument();
 	} );
 
 	test( 'When the scan state is provisioning, show a preparing message', () => {
 		const scanState = { state: 'provisioning' };
 		getSiteScanState.mockReturnValue( scanState );
-		expect( getBenefitsCard().props() ).toHaveProperty( 'jestMarker', 'provisioning' );
+
+		render( <JetpackBenefitsScanHistory /> );
+		const card = screen.getByTestId( 'provisioning' );
+
+		expect( card ).toBeInTheDocument();
 	} );
 
 	test( 'When the scan state is still loading, show a loading placeholder', () => {
 		getSiteScanState.mockReturnValue( null );
 		isRequestingJetpackScan.mockReturnValue( true );
-		expect( getBenefitsCard().props() ).toHaveProperty( 'jestMarker', 'loading' );
+
+		render( <JetpackBenefitsScanHistory /> );
+		const card = screen.getByTestId( 'loading' );
+
+		expect( card ).toBeInTheDocument();
 	} );
 
 	test( 'When the scan state is not present and not loading, show an error', () => {
 		getSiteScanState.mockReturnValue( null );
 		isRequestingJetpackScan.mockReturnValue( false );
-		expect( getBenefitsCard().props() ).toHaveProperty( 'jestMarker', 'error' );
+
+		render( <JetpackBenefitsScanHistory /> );
+		const card = screen.getByTestId( 'error' );
+
+		expect( card ).toBeInTheDocument();
 	} );
 
 	test( 'When the scan state is idle, but there is not a recent scan, indicate that a scan is scheduled', () => {
 		const scanState = { state: 'idle' };
 		getSiteScanState.mockReturnValue( scanState );
 		isRequestingJetpackScan.mockReturnValue( false );
-		expect( getBenefitsCard().props() ).toHaveProperty( 'jestMarker', 'scheduled' );
+
+		render( <JetpackBenefitsScanHistory /> );
+		const card = screen.getByTestId( 'scheduled' );
+
+		expect( card ).toBeInTheDocument();
 	} );
 
 	test( 'If product is standalone scan, the expanded standalone card is rendered', () => {
 		getSiteScanState.mockReturnValue( { state: 'idle', mostRecent: { timestamp: '' } } );
 		isRequestingJetpackScan.mockReturnValue( false );
 
-		const standaloneCard = shallow( <JetpackBenefitsScanHistory isStandalone={ true } /> )
-			.find( 'JetpackBenefitsScanHistory' )
-			.dive()
-			.find( JetpackBenefitsStandaloneCard );
-
-		expect( standaloneCard ).toHaveLength( 1 );
+		const { container } = render( <JetpackBenefitsScanHistory isStandalone={ true } /> );
+		expect( container ).toMatchSnapshot();
```

Comment:
What value does the snapshot test provide? I'd been under the impression we were removing a lot of snapshot tests as well, so I'm curious what indicates one should be removed or added :)

Reply:
For what it's worth, I'm happy to keep them, they add very little overhead. As long as we don't have one for every component, we should be good. They really don't make sense when they're the only test, though, because they create a false sense of security and a false claim that the component is fully covered by tests.

Reply:
Using a snapshot here is purely to remind us of unintended changes, as I haven't found a good way to replace the previous testing strategy without testing too much of the implementation.

Reply:
Thanks for the added context!